classes/Attachment.php

```diff
@@ -147,6 +170,33 @@ public static function getAttachments($idLang, $idProduct, $include = true)
         );
     }
 
+    /**
+     * Unassociate all products from the current object
+     *
+     * @param bool $update_cache If set to true attachment cache will be updated
+     *
+     * @return bool Deletion result
+     */
+    public function deleteAttachments($updateAttachmentCache = true)
+    {
+        $productIds = Db::getInstance()->executeS(
```

Comment:
Shouldn't the function body be wrapped within an `if ((int) $this->id > 0)) { ... }`?